### ISSUE TITLE
Replication of experiment gave different results

### ISSUE THREAD

**Issue by lauravanpoppel** [SakurajimaMaiii/ProtoKD#1]
Hello, 

I trained the teacher, baseline model (for modalitiy 0), and subsequently the protokd  (for modalitiy 0). 

I got the following dice on the test set for the baseline modality 0:
Statistical indicators on test set(wt/co/ec):
Dice:[73.69,65.20,43.07]
HD:[10.98546,11.55212,9.22739]
Sen:[0.80821,0.63665,0.42127]
Spe:[0.99485,0.99866,0.99844]
Average dice is 60.654466581133875.
Average hd is 10.588322678832194.
Average spe is 0.9973164042148.
Average sen is 0.6220434795986626.
Evaluation is finished

And the following for the protokd:
Statistical indicators on test set(wt/co/ec):
Dice:[74.91,67.26,41.59]
HD:[11.56761,12.25499,11.28984]
Sen:[0.79886,0.72309,0.48333]
Spe:[0.99556,0.99805,0.99710]
Average dice is 61.252319872711674.
Average hd is 11.704147892614456.
Average spe is 0.9969007474946627.
Average sen is 0.6684268171434242.
Evaluation is finished

These are quite different from the ones reported in the paper. 
Do you have any idea why?

Best,

**Comment by SakurajimaMaiii** [comment]
Hi, **what is the performance of your teacher model** ?
The baseline (unimodal) achieves 60.65 Dice. I think it is normal.
The low  performance of protokd may casued by bad teacher model .
May I know the performance of teacher model (trained by yourself)? I think  ~80 dice is OK for teacher model.

**Comment by lauravanpoppel** [comment]
Hello, thanks for you fast response. I think i exactly replicated your experiment on the brats2018 data. I ran your script and did not adjust anything (using all default parameters). So I'm curious why my results differ from yours presented in the paper. 

I havent validated the teacher model on the test set but these were the results of the teacher after training/validation:
Training finished, tensorboardX writer closed
Best epoch is 676, best mean dice is 0.867973
Dice of wt/co/ec is 0.885766,0.888012,0.830141
Training total time: 24.438457 hours.

These were the results of the baseline model (modality 0) on the validation set:
Training finished, tensorboardX writer closed
Best epoch is 877, best mean dice is 0.711322
Dice of wt/co/ec is 0.807553,0.767383,0.559028
Training total time: 22.842150 hours.

These were the results of the protokd model (modality 0) on the validation set:
Training finished, tensorboardX writer closed
Best epoch is 721, best mean dice is 0.693032
Dice of wt/co/ec is 0.808849,0.762912,0.507336
Training total time: 21.617641 hours.

**Comment by lucasdevries** [comment]
@SakurajimaMaiii I have the same issue... Are all default hyperparameters set similarly as in the paper? Same data augmentations, etc?
My test set Dice for Unimodal T1 is 73.69, 65.20, and 43.07. Mean 60.65. In the paper, it is 58.77 average. Any idea what causes the difference?

**Comment by SakurajimaMaiii** [comment]
Sorry for the late reply. Due to the relatively small dataset, the results may have higher variability. I will upload our trained model later.

**Comment by SakurajimaMaiii** [comment]
@lucasdevries  @lauravanpoppel  
I have uploaded some trained models, including teacher, baseline T1, protokd T1 in [Google Drive](https://drive.google.com/drive/folders/1DhCBMn5Z002TzsfRwFzu_pvXjYc4BoCn?usp=sharing).
